aiida/cmdline/commands/cmd_code.py

```diff
@@ -250,7 +250,8 @@ def export(code, output_file):
         else:
             value = getattr(code, key)
 
-        code_data[key] = str(value)
+        if value is not None:
```

Comment:
It is not to keep the YAML compact. I have added a comment for the real reason:
```
        # If the attribute is not set, for example ``with_mpi`` do not export it, because the YAML won't be valid for
        # use in ``verdi code create`` since ``None`` is not a valid value on the CLI.
```